RestaurantReservationSystem/EntityServices/MenuItemService.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.Models;
+using RestaurantReservation.Db.Repositories.Interfaces;
+using RestaurantReservation.Db.Services.Interfaces;
+using RestaurantReservationSystem.Constants;
+
+public class MenuItemService : IMenuItemService
+{
+    private readonly IMenuItemRepository _menuItemRepository;
+
+    public MenuItemService(IMenuItemRepository menuItemRepository)
+    {
+        _menuItemRepository = menuItemRepository;
+    }
+
+    public async Task AddMenuItemAsync(int restaurantId, string name, string description, decimal price)
+    {
+        var menuItem = new MenuItem
+        {
+            RestaurantId = restaurantId,
+            Name = name,
+            Description = description,
+            Price = price
+        };
+
+        try
+        {
+            await _menuItemRepository.AddAsync(menuItem);
+        }
+        catch (Exception ex)
+        {
+            throw new InvalidOperationException(DefaultErrorMessages.AddFailed, ex);
+        }
+    }
+
+    public async Task GetAllMenuItemsAsync()
+    {
+        try
+        {
+            var allItems = await _menuItemRepository.GetAllAsync();
+            foreach (var item in allItems)
+            {
+                Console.WriteLine($"[MenuItem] {item.ItemId} - Name: {item.Name}, Price: {item.Price}, RestaurantId: {item.RestaurantId}");
+            }
+        }
+        catch (Exception ex)
+        {
+            throw new InvalidOperationException(DefaultErrorMessages.RetrieveFailed, ex);
+        }
+    }
+
+    public async Task UpdateMenuItemAsync(int itemId, int UpdatedrestaurantId, string Updatedname,
+        string Updateddescription, decimal Updatedprice)
+    {
+        var updatedItem = new MenuItem
+        {
+            ItemId = 1, 
```

Comment:
we shouldnt be hardcoding the ids, either you pass them or the user pass the item name we lookup db for the id and use it